libraries/Microsoft.Bot.Configuration/BotConfiguration.cs

```diff
@@ -271,9 +316,9 @@ public void DisconnectService(string id)
         }
 
         /// <summary>
-        ///  make sure secret is correct by decrypting the secretKey with it
+        /// Make sure secret is correct by decrypting the secretKey with it.
         /// </summary>
-        /// <param name="secret">secret to use</param>
+        /// <param name="secret">Secret to use.</param>
```

Comment:
The behavior here is really weird, in that a null string is valid.

Reply:
If secret is null, it means that no decryption is needed. Yes, this class mixes 1) object model with 2) storage (i.e. strongly coupled to file system) and 3) encryption. 3 responsibilities -> we should likely have 3 classes, this is not SOLID :)